packages/mask/src/plugins/WalletService/services/wallet/database/network.ts

```diff
@@ -0,0 +1,72 @@
+import { asyncIteratorToArray } from '@masknet/shared-base'
+import { omit } from 'lodash-es'
+import { PluginDB } from '../../../database/Plugin.db.js'
+
+interface Network {
+    name: string
+    chainId: number
+    rpc: string
+    currencySymbol?: string
+    explorer?: string
+}
+
+export interface NetworkRecord extends Network {
+    type: 'network'
+    /** chain id as id */
+    id: number
+    createdAt: number
+    updatedAt: number
+}
+
+function recordToNetwork(record: NetworkRecord): Network {
+    return omit(record, 'id', 'type', 'createdAt', 'updatedAt')
+}
+
+export async function getNetworks() {
+    const networks = (await asyncIteratorToArray(PluginDB.iterate('network'))).map((x) => x.value)
+
+    return networks
+        .sort((a, z) => {
+            if (a.updatedAt > z.updatedAt) return -1
+            if (a.updatedAt < z.updatedAt) return 1
+            if (a.createdAt > z.createdAt) return -1
+            if (a.createdAt < z.createdAt) return 1
+            return 0
+        })
+        .map(recordToNetwork)
+}
+
+export async function getNetwork(chainId: number) {
+    if (!chainId) return null
+    const record = (await PluginDB.get('network', chainId)) ?? null
+    return record ? recordToNetwork(record) : null
+}
+
+export async function addNetwork(network: Network) {
+    const now = Date.now()
+    await PluginDB.add({
```

Comment:
Some validation rules:

1. The chain id must be a position integer.
2. The chain id not intersect with any supported network.
3. Validate the network configuration by data from `https://chainlist.wtf/`.
4. The provider URL required HTTPS.

Reply:
> The provider URL required HTTPS.

127.0.0.1 exception?

Reply:
> The chain id not intersect with any supported network.

PM said we duplicated chain ids are allowed.

> The provider URL required HTTPS.

No http? If a RPC request success, we can assume the RPC is available, can't we?